application/applicationextension.inc.php

```diff
@@ -1709,8 +1709,16 @@ public static function GetObjectSetFromKey($sClass, $key, $iLimit = 0, $iOffset
 		elseif (is_string($key))
 		{
 			// OQL
-			$oSearch = DBObjectSearch::FromOQL($key);
-		}
+            try {
+                $oSearch = DBObjectSearch::FromOQL($key);
+            } catch (Exception $e) {
+                throw new CoreOqlException('json_data key parameter: query failed to execute', [
```

Comment:
I think a bit shorter message is fine as well:
```suggestion
                throw new CoreOqlException('Query failed to execute', [
```

Reply:
This is an option indeed
Now you can be really confused about the message (it is an iTop error, does one of my parameters triggers it ?)
I chose a detailed error message so that rest.php user will now exactly what to fix, but stating "the query inside key param in json_data" might be a bit much.
We'll see during the reviews.

Reply:
Hello,
During review we decided to go on with your proposition , as the throw is in a method that doesn't know about the json_data structure, and this message is simple and gives the information needed by the consumer.